state.go

```diff
@@ -129,8 +137,19 @@ type StateRunner interface {
 	// Obtain runner's state
 	GetState() State
 	// TODO generic state run action
+	RunState(ctx *StateContext) (State, bool)
+}
+
+type StateData struct {
+	Update UpdateResponse
```

Comment:
`State` name is already taken.

Also `LastUpdateInfo` might be a bit confusing. When is the update cosidered to be last? For instance in commit state, update info is technically not the last update but for the current one.